python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -388,89 +404,293 @@ def _convolve(self, exposure: afwImage.Exposure, modelPsf: afwDetection.Gaussian
-                            aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0,
-                                                           normalize=True)
-                            fluxScaling = 0.5*sigma**2/aperSigma2
-                    except (InvalidParameterError, ZeroDivisionError):
-                        # Raised when the aperture is invalid
-                        flagKey = measRecord.schema.join(baseName, "flag_bigpsf")
-                        measRecord.set(flagKey, 1)
-                        continue
-
-                fluxResult = measBase.SdssShapeAlgorithm.computeFixedMomentsFlux(convolved, aperShape, center)
-
-                # Calculate the integral in Eq. A17 of Kuijken et al. (2015)
-                # ``fluxErrScaling`` contains the factors not captured by
-                # ``fluxScaling`` and `instFluxErr`. It is 1 theoretically
-                # if ``kernelAcf`` is a Dirac-delta function.
-                fluxErrScaling = self._getFluxErrScaling(kernelAcf, aperShape)
-
-                # Scale the fluxResult and copy result to record
-                fluxResult.instFlux *= fluxScaling
-                fluxResult.instFluxErr *= fluxScaling*fluxErrScaling
-                fluxResultKey = FluxResultKey(measRecord.schema[baseName])
-                fluxResultKey.set(measRecord, fluxResult)
+            measureFlux = partial(self._measureFlux, measRecord, convolved, kernelAcf, center)
+
+            if self.config.doPsfPhotometry:
+                baseName = self.ConfigClass._getGaapResultName(scalingFactor, "PsfFlux", self.name)
+                aperShape = targetPsf.computeShape()
+                measureFlux(aperShape, baseName, fluxScaling=1)
+
+            # Iterate over pre-defined circular apertures
+            for sigma in self.config.sigmas:
+                baseName = self.ConfigClass._getGaapResultName(scalingFactor, sigma, self.name)
+                if sigma <= targetSigma:
+                    # Raise when the aperture is invalid
+                    self._setFlag(measRecord, baseName)
+                    continue
+
+                aperSigma2 = sigma**2 - targetSigma**2
+                aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0)
+                fluxScaling = 0.5*sigma**2/aperSigma2
+                measureFlux(aperShape, baseName, fluxScaling)
 
         # Raise GaapConvolutionError before exiting the plugin
         # if the collection of errors is not empty
         if errorCollection:
             raise GaapConvolutionError(errorCollection)
 
+    @staticmethod
+    def _setFlag(measRecord, baseName):
+        """Set the `bigpsf` flag.
+
+        Parameters
+        ----------
+        measRecord : `~lsst.afw.table.SourceRecord`
+            Record describing the source being measured.
+        baseName : `str`
+            The base name of the GAaP field for which the flag must be set.
+        """
+        flagKey = measRecord.schema.join(baseName, "flag_bigpsf")
```

Comment:
Ah, sorry, I was going through commits one after the other. This actually makes perfect sense.

I would still suggest changing the name of the function to something less general than `_setFlag`!

Reply:
When I came up with flag definitions, I wasn't sure if I had exhausted all possible flags. Now I'm fairly certain I wouldn't need more flags and I can make this just `flag` instead of `flag_bigpsf`.

Reply:
It's always good to leave some room for future flags without having to change the meaning of old ones.

Reply:
I've kept the flag name and the function name the same, but allow `flagName` to be passed (which is "bigpsf" by default now). So the generic function name is justified while the use case is specific to one flag.

Reply:
Coming back on flag suffixes, during the aperture correction stage, it expects the flag column to end with `_flag`. Currently, this is the only flag column for each `instFlux`. I can either rename this to `_flag` instead of `_flag_bigpsf` and set any other failure modes in the future to also set the `_flag`, or have a separate `_flag` field that is set if any of the flags (currently only `_flag_bigpsf`) is set. Thoughts?

Reply:
Preferred pattern is definitely the latter: have both a generic flag and specific flags, even if there is only one of the later to start with.

Reply:
Thanks Jim. I was worried about proliferation of flag fields, but we're only talking about increasing it from 3 to 6 with the default settings, which is not bad. I'll incorporate that in the aperture correction branch then.